File: LmsApi/Models/Course.cs
namespace LmsApi.Models;

public class Course
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string Category { get; set; } = string.Empty;
    public string Subcategory { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public decimal OriginalPrice { get; set; }
    public int Discount { get; set; }
    public string Badge { get; set; } = string.Empty;
    public double Rating { get; set; }
    public int ReviewCount { get; set; }
    public int StudentCount { get; set; }
    public int InstructorId { get; set; }
    public Instructor? Instructor { get; set; }
    public string ImageUrl { get; set; } = string.Empty;
    public List<string> WhatYouLearn { get; set; } = new();
    public List<string> Includes { get; set; } = new();
    public List<string> Companies { get; set; } = new();
    public List<CourseSection> CourseSections { get; set; } = new();
    public List<CourseReview> Reviews { get; set; } = new();
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

public class Instructor
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public double Rating { get; set; }
    public int Students { get; set; }
    public int Courses { get; set; }
    public string Bio { get; set; } = string.Empty;
    public string ImageUrl { get; set; } = string.Empty;
    public List<Course> CreatedCourses { get; set; } = new();
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

public class CourseSection
{
    public int Id { get; set; }
    public int CourseId { get; set; }
    public Course? Course { get; set; }
    public string Day { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Duration { get; set; } = string.Empty;
    public List<CourseLecture> Lectures { get; set; } = new();
}

public class CourseLecture
{
    public int Id { get; set; }
    public int SectionId { get; set; }
    public CourseSection? Section { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Duration { get; set; } = string.Empty;
    public string? VideoUrl { get; set; }
}

public class CourseReview
{
    public int Id { get; set; }
    public int CourseId { get; set; }
    public Course? Course { get; set; }
    public string Name { get; set; } = string.Empty;
    public int Rating { get; set; }
    public string Time { get; set; } = string.Empty;
    public string Text { get; set; } = string.Empty;
}

File: web/src/types/course.ts
export interface Course {
  id: number;                 // ✅ FIXED
  title: string;
  description: string;
  category: string;
  subcategory: string;

  price: number;
  originalPrice: number;
  discount: number;

  badge: string;
  rating: number;
  reviewCount: number;
  studentCount: number;

  instructorId: number;       // ✅ FIXED
  instructor?: {
    id: number;
    name: string;
    title: string;
    rating: number;
    students: number;
    courses: number;
    bio: string;
    imageUrl: string;
  };

  imageUrl: string;           // ✅ NOW RESOLVES FOR ALL CARDS

  whatYouLearn: string[];
  includes: string[];
  companies: string[];

  courseSections: CourseSection[]; // ✅ MATCH API
  reviews: CourseReview[];

  createdAt: string;
  updatedAt: string;
}


export interface CourseSection {
  id: number;
  courseId: number;
  day: string;
  title: string;
  duration: string;
  lectures: CourseLecture[];   // ✅ MATCH API
}

export interface CourseLecture {
  id: number;
  sectionId: number;
  title: string;
  duration: string;
  videoUrl?: string;
}


export interface CourseReview {
  id?: number;
  courseId?: number;
  name: string;
  rating: number;
  time: string;
  text: string;
}

export interface CreateCourseInput {
  title: string;
  description: string;
  category: string;
  subcategory: string;
  price: number;
  originalPrice: number;
  badge: string;
  instructorId: number;
  imageUrl: string;
  whatYouLearn: string[];
  includes: string[];
  companies?: string[];
}
